File: BinaryTreeInOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.nodes = []
    
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root is None:
            return self.nodes
        
        self.inorderTraversal(root.left)
        self.nodes.append(root.val)
        self.inorderTraversal(root.right)
        return self.nodes

File: BinaryTreeInOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> nodes;
    void Traverse(TreeNode* node) {
        if (node == nullptr) return;
        Traverse(node->left);
        nodes.push_back(node->val);
        Traverse(node->right);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        Traverse(root);
        return nodes;
    }
};
